opm/material/fluidmatrixinteractions/EclDefaultMaterial.hpp

```diff
@@ -325,18 +325,21 @@ class EclDefaultMaterial : public TraitsT
     {
         typedef MathToolbox<typename FluidState::Scalar> FsToolbox;
 
-        Scalar Swco = params.Swl();
-
-        Scalar Sw = FsToolbox::value(fluidState.saturation(waterPhaseIdx));
+        Scalar So = FsToolbox::value(fluidState.saturation(oilPhaseIdx));
         Scalar Sg = FsToolbox::value(fluidState.saturation(gasPhaseIdx));
-        Sw = std::min(1.0, std::max(Swco, Sw));
         Sg = std::min(1.0, std::max(0.0, Sg));
 
-        Scalar Sw_ow = Sg + Sw;
-        Scalar So_go = 1 - Sw_ow + Swco;
-
-        params.oilWaterParams().update(/*pcSw=*/Sw, /*krwSw=*/1 - Sg, /*krnSw=*/Sw_ow);
-        params.gasOilParams().update(/*pcSw=*/1 - Sg, /*krwSw=*/So_go, /*krnSw=*/1 - Sg);
+        // FIXME: the saturations which are passed to update the hysteresis curves are
+        // inconsistent with the ones used to calculate the relative permabilities. We do
+        // it like this anyway because (a) the saturation functions of opm-core do it
+        // this way (b) the simulations seem to converge better (which is not too much
+        // surprising actually, because the time step does not start on a kink in the
+        // solution) and (c) the Eclipse 100 simulator may do the same.
+        //
+        // Though be aware that from a physical perspective this is definitively
+        // incorrect!
+        params.oilWaterParams().update(/*pcSw=*/1 - So, /*krwSw=*/1 - So, /*krn_Sw=*/1 - So);
```

Comment:
> sure: a runtime option would have a (probably small) influence on performance, a template parameter or a value which is hard-coded into the parameters would be less flexible. I think I slightly prefer the compile time option, but you choose...

I prefer the runtime choice unless it costs too much performance (and in this case, if it is measurably costly we could move the `if` to the routine that contains the for-loop perhaps). These days I tend to think that the compile-time option is too inflexible for users to ever get used unless it contains major differences such as a solvent model, but then we tend to make a separate program anyway.